### PR TITLE
Improve a bit the documentation of Accept and add sorting.

### PR DESCRIPTION
**Pull request by Drup**
We could put the q-stuff in a submodule. Not sure if necessary.

### CODE REVIEW

**Comment by rgrinberg** [comment]
* I'd merge the docs improvement immediately if it was in a different commit

* The implementation in this PR obviously looks good but since I know next to nothing about this "quality" stuff. Therefore I will leave this to @dsheets

File: lib/accept.mli [hunk 1]
```diff
@@ -43,13 +56,14 @@ type encoding =
   | Identity
   | AnyEncoding with sexp
 
+(** Language tags, as a list of subtags.
+    ["en-gb"] is represented as [Language ["en"; "gb"]].
+    @see <http://www.w3.org/Protocols/rfc2616/rfc2616-sec14.html#sec14.4> The spec.
+*)
 type language = Accept_types.language =
```
Comment (dbuenzli) [inline lib/accept.mli]: Note that this is formally a BCP47 *basic language range* tag and not a *language tag* (FWIW I do have a module [here](https://github.com/dbuenzli/remat/blob/master/src-common/nlang.mli) to handle these that I expect to release as an independent module at some point).  Also referring to https://tools.ietf.org/html/rfc7231#section-5.3.5 would be nicer, the latter obsoletes 2616 and makes a lot of things more clear notably in this case it links to the right RFC.

Comment (Drup) [inline lib/accept.mli]: I know about BCP47, I was also considering doing such a module, happy to see you did it already.

Thanks for the link. It's updated. Is there a guideline in mirage libraries on how to reference RFCs and so on ? I guess it's waiting for ppx and [reqtrace](https://github.com/infidel/reqtrace) ?

**Comment by Drup** [comment]
@rgrinberg done.

**Comment by rgrinberg** [comment]
@Drup applied first patch, thanks.

**Comment by dsheets** [comment]
LGTM.

**Comment by rgrinberg** [comment]
@Drup please rebase

**Comment by Drup** [comment]
Already did 2 minutes ago :D

**Pull request merged**